admin/class-activation-redirect.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Handles plugin activation redirect to welcome page.
+ *
+ * @package Accessibility_Checker\Admin
+ */
+
+namespace EDAC\Admin;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Class Activation_Redirect
+ *
+ * Handles redirecting to the welcome page after plugin activation.
+ *
+ * @since 1.35.0
+ */
+class Activation_Redirect {
+
+	/**
+	 * The page slug for the welcome page.
+	 *
+	 * @since 1.35.0
+	 * @var string
+	 */
+	const WELCOME_PAGE_SLUG = 'accessibility_checker';
+
+	/**
+	 * Initialize the activation redirect.
+	 *
+	 * @since 1.35.0
+	 */
+	public function init(): void {
+		add_action( 'admin_init', [ $this, 'maybe_redirect_to_welcome' ] );
+	}
+
+	/**
+	 * Get the welcome page URL.
+	 *
+	 * @since 1.35.0
+	 * @return string The URL to the welcome page.
+	 */
+	public function get_welcome_page_url(): string {
+		return admin_url( 'admin.php?page=' . self::WELCOME_PAGE_SLUG );
+	}
+
+	/**
+	 * Redirect to welcome page after activation if conditions are met.
+	 *
+	 * This will only redirect if:
+	 * - The activation redirect transient is set
+	 * - We're not doing an AJAX request
+	 * - We're not in the network admin (multisite)
+	 * - We're not activating multiple plugins at once
+	 * - User has permission to access the welcome page
+	 *
+	 * @since 1.35.0
+	 * @return void
+	 */
+	public function maybe_redirect_to_welcome(): void {
+		// Check if the activation redirect transient exists.
+		if ( ! get_transient( 'edac_activation_redirect' ) ) {
+			return;
+		}
+
+		// Delete the transient immediately to prevent redirect loops.
+		delete_transient( 'edac_activation_redirect' );
```

Comment:
This would be better placed directly before the redirect instead of before all of these extra checks and early returns.